lib/PluginFindRecord/PluginFindRecordModal.js

```diff
@@ -0,0 +1,402 @@
+                      <Pane
+                        defaultWidth="22%"
+                        paneTitle={<FormattedMessage id="stripes-smart-components.searchAndFilter" />}
+                      >
+                        <form onSubmit={onSubmitSearch}>
+                          <div className={css.searchGroupWrap}>
+                            <SearchField
+                              autoFocus
+                              className={css.searchField}
+                              data-test-plugin-search-input
+                              marginBottom0
+                              name="query"
+                              onChange={getSearchHandlers().query}
+                              onClear={getSearchHandlers().reset}
+                              value={searchValue.query}
+                            />
+                            <Button
+                              buttonStyle="primary"
+                              data-test-plugin-search-submit
+                              disabled={(!searchValue.query || searchValue.query === '')}
+                              fullWidth
+                              marginBottom0
+                              type="submit"
+                            >
+                              <FormattedMessage id="stripes-smart-components.search" />
+                            </Button>
+                          </div>
+                          <div className={css.resetButtonWrap}>
+                            <Button
+                              buttonStyle="none"
+                              disabled={disableReset()}
+                              fullWidth
+                              id="clickable-reset-all"
+                              onClick={resetAll}
+                            >
+                              <Icon icon="times-circle-solid">
+                                <FormattedMessage id="stripes-smart-components.resetAll" />
+                              </Icon>
+                            </Button>
+                          </div>
+                          {renderFilters
+                            ? renderFilters(activeFilters.state, getFilterHandlers())
+                            : (
+                              <Filters
+                                activeFilters={activeFilters}
+                                config={filterConfig}
+                                onChangeHandlers={getFilterHandlers()}
+                              />
+                            )
+                          }
+                        </form>
+                      </Pane>
+                    }
+                    <Pane
+                      defaultWidth="fill"
+                      firstMenu={this.renderResultsFirstMenu(activeFilters)}
+                      padContent={false}
+                      paneSub={resultPaneSub}
+                      paneTitle={RESULTS_HEADER}
+                    >
+                      <MultiColumnList
```

Comment:
This might fall under 'future work' for this component, but I would take the Search field, and the contents of the  "results" pane, and wrap them both in render functions that can be overridden by props. As @doytch said, we've been down this road with SearchAndSort for prettymuch the same reasons (reduce the boilerplate.) If a find-record plugin needs to differentiate (ie, have a different interface for search fields - see something like the tabbed search UI for eHoldings or for ui-greements) it's only left with the option to fork this and adjust it locally. The use of this is pretty limited to 'find this one record' but it has to be almost as flexible as SearchAndSortQuery itself - with the healthy defaults to get us over through the UI boilerplate this is meant to help.

Reply:
John, mostly I agree with you at the high level. But can we do it after the release?
The main point of this PR - is to fix ui-plugin-find-* that are broken on search and filtering before 3.2 release. I've just leveraged code, that was copy-pasted across plugins, and the code, you've posted into ui-plugin-find-user. If I had more time I would love to provide some solid and better solution, but as we are short on time and we have a couple ui-plugins to fix - I don't think it's a good call to do it before release. I think the main point not to break existing functionality in the release.

Reply:
You're both right; we need more flexibility in the long term _and_ we have a bug to squash in the short term. I think your "future work" comment is spot-on, @JohnC-80. 

As long as the current implementation doesn't cut off avenues for future deployment, i.e. we can extend the API later with render props but leave the current API in place, I think we should commit this as-is. Agreed?

Reply:
Yep. Let's just ear-mark the idea of override patterns in the component. No need to implement that immediately.